### DISCUSSION TITLE
table layout with shadow dom

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#3478] [Q&A]
When make something like "Table Parent Component + Table Item Component".
display:table and display:table-row don't work correctly. 

```html
<table-parent-component>
  <table-row-component></table-row-component>
  <table-row-component></table-row-component>
  <table-row-component></table-row-component>
  <table-row-component></table-row-component>
  <table-row-component></table-row-component>
</table-parent-component>
```

I just solve this problem by using createRenderRoot.
like this

```html
<table-parent-component>
  <table-row-component style="display:table-row;">//html. no shadow dom</table-row-component>
  <table-row-component style="display:table-row;">//html. no shadow dom</table-row-component>
  <table-row-component style="display:table-row;">//html. no shadow dom</table-row-component>
  <table-row-component style="display:table-row;">//html. no shadow dom</table-row-component>
  <table-row-component style="display:table-row;">//html. no shadow dom</table-row-component>
</table-parent-component>
```

how can i make something above?
there are any good idea?

related issues
https://stackoverflow.com/a/65303542/12365658

**Comment by bicknellr** [comment]
Could you give more detail on what you were expecting those `display` values to do that they aren't? Or a repro if possible? [Here's an example of a table built with `display`](https://lit.dev/playground/#project=W3sibmFtZSI6ImV4YW1wbGUtdGFibGUuanMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBjc3MsIExpdEVsZW1lbnR9IGZyb20gJ2xpdCc7XG5cbmV4cG9ydCBjbGFzcyBFeGFtcGxlVGFibGUgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BcbiAgICA6aG9zdCB7XG4gICAgICBkaXNwbGF5OiB0YWJsZTtcbiAgICB9XG4gIGA7XG5cbiAgY29ubmVjdGVkQ2FsbGJhY2soKSB7XG4gICAgc3VwZXIuY29ubmVjdGVkQ2FsbGJhY2soKTtcbiAgICB0aGlzLnNldEF0dHJpYnV0ZSgncm9sZScsICd0YWJsZScpO1xuICB9XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxzbG90Pjwvc2xvdD5gO1xuICB9XG59XG5jdXN0b21FbGVtZW50cy5kZWZpbmUoJ2V4YW1wbGUtdGFibGUnLCBFeGFtcGxlVGFibGUpOyJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9leGFtcGxlLXRhYmxlLmpzXCI-PC9zY3JpcHQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vZXhhbXBsZS10aGVhZC5qc1wiPjwvc2NyaXB0PlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL2V4YW1wbGUtdGJvZHkuanNcIj48L3NjcmlwdD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9leGFtcGxlLXRyLmpzXCI-PC9zY3JpcHQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vZXhhbXBsZS10ZC5qc1wiPjwvc2NyaXB0PlxuICA8c3R5bGU-XG4gICAgZXhhbXBsZS10aGVhZCB7XG4gICAgICBiYWNrZ3JvdW5kLWNvbG9yOiBsaWdodGJsdWU7XG4gICAgfVxuICAgIFxuICAgIGV4YW1wbGUtdGJvZHkgPiBleGFtcGxlLXRyOm50aC1jaGlsZCgybikge1xuICAgICAgYmFja2dyb3VuZDogbGlnaHRncmF5O1xuICAgIH1cbiAgPC9zdHlsZT5cbjwvaGVhZD5cbjxib2R5PlxuICA8ZXhhbXBsZS10YWJsZT5cbiAgICA8ZXhhbXBsZS10aGVhZD5cbiAgICAgIDxleGFtcGxlLXRyPlxuICAgICAgICA8ZXhhbXBsZS10ZD5mcnVpdDwvZXhhbXBsZS10ZD5cbiAgICAgICAgPGV4YW1wbGUtdGQ-Y29sb3I8L2V4YW1wbGUtdGQ-XG4gICAgICA8L2V4YW1wbGUtdHI-XG4gICAgPC9leGFtcGxlLXRoZWFkPlxuICAgIDxleGFtcGxlLXRib2R5PlxuICAgICAgPGV4YW1wbGUtdHI-XG4gICAgICAgIDxleGFtcGxlLXRkPm9yYW5nZTwvZXhhbXBsZS10ZD5cbiAgICAgICAgPGV4YW1wbGUtdGQ-b3JhbmdlPC9leGFtcGxlLXRkPlxuICAgICAgPC9leGFtcGxlLXRyPlxuICAgICAgPGV4YW1wbGUtdHI-XG4gICAgICAgIDxleGFtcGxlLXRkPmJhbmFuYTwvZXhhbXBsZS10ZD5cbiAgICAgICAgPGV4YW1wbGUtdGQ-eWVsbG93PC9leGFtcGxlLXRkPlxuICAgICAgPC9leGFtcGxlLXRyPlxuICAgICAgPGV4YW1wbGUtdHI-XG4gICAgICAgIDxleGFtcGxlLXRkPmxpbWU8L2V4YW1wbGUtdGQ-XG4gICAgICAgIDxleGFtcGxlLXRkPmdyZWVuPC9leGFtcGxlLXRkPlxuICAgICAgPC9leGFtcGxlLXRyPlxuICAgIDwvZXhhbXBsZS10Ym9keT5cbiAgPC9leGFtcGxlLXRhYmxlPlxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9LHsibmFtZSI6ImV4YW1wbGUtdHIuanMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBjc3MsIExpdEVsZW1lbnR9IGZyb20gJ2xpdCc7XG5cbmV4cG9ydCBjbGFzcyBFeGFtcGxlVHIgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BcbiAgICA6aG9zdCB7XG4gICAgICBkaXNwbGF5OiB0YWJsZS1yb3c7XG4gICAgfVxuICBgO1xuXG4gIGNvbm5lY3RlZENhbGxiYWNrKCkge1xuICAgIHN1cGVyLmNvbm5lY3RlZENhbGxiYWNrKCk7XG4gICAgdGhpcy5zZXRBdHRyaWJ1dGUoJ3JvbGUnLCAncm93Jyk7XG4gIH1cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgPHNsb3Q-PC9zbG90PmA7XG4gIH1cbn1cbmN1c3RvbUVsZW1lbnRzLmRlZmluZSgnZXhhbXBsZS10cicsIEV4YW1wbGVUcik7In0seyJuYW1lIjoiZXhhbXBsZS10ZC5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcblxuZXhwb3J0IGNsYXNzIEV4YW1wbGVUZCBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgIDpob3N0IHtcbiAgICAgIGRpc3BsYXk6IHRhYmxlLWNlbGw7XG4gICAgICBwYWRkaW5nOiAwLjI1ZW07XG4gICAgfVxuICBgO1xuXG4gIGNvbm5lY3RlZENhbGxiYWNrKCkge1xuICAgIHN1cGVyLmNvbm5lY3RlZENhbGxiYWNrKCk7XG4gICAgdGhpcy5zZXRBdHRyaWJ1dGUoJ3JvbGUnLCAnY2VsbCcpO1xuICB9XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxzbG90Pjwvc2xvdD5gO1xuICB9XG59XG5jdXN0b21FbGVtZW50cy5kZWZpbmUoJ2V4YW1wbGUtdGQnLCBFeGFtcGxlVGQpOyJ9LHsibmFtZSI6ImV4YW1wbGUtdGhlYWQuanMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBjc3MsIExpdEVsZW1lbnR9IGZyb20gJ2xpdCc7XG5cbmV4cG9ydCBjbGFzcyBFeGFtcGxlVGhlYWQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BcbiAgICA6aG9zdCB7XG4gICAgICBkaXNwbGF5OiB0YWJsZS1oZWFkZXItZ3JvdXA7XG4gICAgfVxuICBgO1xuXG4gIGNvbm5lY3RlZENhbGxiYWNrKCkge1xuICAgIHN1cGVyLmNvbm5lY3RlZENhbGxiYWNrKCk7XG4gICAgdGhpcy5zZXRBdHRyaWJ1dGUoJ3JvbGUnLCAncm93Z3JvdXAnKTtcbiAgfVxuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGA8c2xvdD48L3Nsb3Q-YDtcbiAgfVxufVxuY3VzdG9tRWxlbWVudHMuZGVmaW5lKCdleGFtcGxlLXRoZWFkJywgRXhhbXBsZVRoZWFkKTsifSx7Im5hbWUiOiJleGFtcGxlLXRib2R5LmpzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgY3NzLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuXG5leHBvcnQgY2xhc3MgRXhhbXBsZVRib2R5IGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBzdHlsZXMgPSBjc3NgXG4gICAgOmhvc3Qge1xuICAgICAgZGlzcGxheTogdGFibGUtcm93LWdyb3VwO1xuICAgIH1cbiAgYDtcblxuICBjb25uZWN0ZWRDYWxsYmFjaygpIHtcbiAgICBzdXBlci5jb25uZWN0ZWRDYWxsYmFjaygpO1xuICAgIHRoaXMuc2V0QXR0cmlidXRlKCdyb2xlJywgJ3Jvd2dyb3VwJyk7XG4gIH1cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgPHNsb3Q-PC9zbG90PmA7XG4gIH1cbn1cbmN1c3RvbUVsZW1lbnRzLmRlZmluZSgnZXhhbXBsZS10Ym9keScsIEV4YW1wbGVUYm9keSk7In0seyJuYW1lIjoiZXhhbXBsZS10aC5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcblxuZXhwb3J0IGNsYXNzIEV4YW1wbGVUaCBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgIDpob3N0IHtcbiAgICAgIGRpc3BsYXk6IHRhYmxlLWNlbGw7XG4gICAgICBwYWRkaW5nOiAwLjI1ZW07XG4gICAgfVxuICBgO1xuXG4gIGNvbm5lY3RlZENhbGxiYWNrKCkge1xuICAgIHN1cGVyLmNvbm5lY3RlZENhbGxiYWNrKCk7XG4gICAgdGhpcy5zZXRBdHRyaWJ1dGUoJ3JvbGUnLCAnY29sdW1uaGVhZGVyJyk7XG4gIH1cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgPHNsb3Q-PC9zbG90PmA7XG4gIH1cbn1cbmN1c3RvbUVsZW1lbnRzLmRlZmluZSgnZXhhbXBsZS10aCcsIEV4YW1wbGVUaCk7In1d). As far as I can tell, the styles seem to be working.

**Reply by dbwodlf3** [reply]
exactly same. but this example is simple. it works.
yours example is good. but i just wanna more complex.

my example is this. but this is doesn't use shadow dom.
[example](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcblxuQGN1c3RvbUVsZW1lbnQoJ2V4YW1wbGUtcm93LWl0ZW0nKVxuZXhwb3J0IGNsYXNzIEV4YW1wbGVSb3dJdGVtIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBzdHlsZXMgPSBjc3NgcCB7IGNvbG9yOiBibHVlIH1gO1xuXG4gIEBwcm9wZXJ0eSgpXG4gIGNvbnRlbnROYW1lID0gJ1NvbWVib2R5JztcbiAgQHByb3BlcnR5KClcbiAgdmlld3MgPSAwO1xuICBAcHJvcGVydHkoKVxuICB3cml0ZXIgPSBcIlwiO1xuICBAcHJvcGVydHkoKVxuICBjcmVhdGlvblRpbWUgPSBcIlwiO1xuICBcbiAgY3JlYXRlUmVuZGVyUm9vdCgpe1xuICAgIHRoaXMuc3R5bGUuZGlzcGxheSA9IFwidGFibGUtcm93XCI7XG4gICAgcmV0dXJuIHRoaXM7XG4gIH1cbiAgXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgICA8ZGl2IHN0eWxlPVwiZGlzcGxheTp0YWJsZS1jZWxsXCI-JHt0aGlzLmNvbnRlbnROYW1lfTwvZGl2PlxuICAgICAgICAgIDxkaXYgc3R5bGU9XCJkaXNwbGF5OnRhYmxlLWNlbGxcIj4ke3RoaXMudmlld3N9PC9kaXY-XG4gICAgICAgICAgPGRpdiBzdHlsZT1cImRpc3BsYXk6dGFibGUtY2VsbFwiPiR7dGhpcy53cml0ZXJ9PC9kaXY-XG4gICAgICAgICAgPGRpdiBzdHlsZT1cImRpc3BsYXk6dGFibGUtY2VsbFwiPiR7dGhpcy5jcmVhdGlvblRpbWV9PC9kaXY-XG4gICAgYDtcbiAgfVxufVxuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxkaXYgc3R5bGU9XCJkaXNwbGF5OnRhYmxlOyB3aWR0aDoxMDAlO1wiPlxuICAgIDxkaXYgc3R5bGU9XCJkaXNwbGF5OnRhYmxlLXJvd1wiPlxuICAgICAgPGRpdiBzdHlsZT1cImRpc3BsYXk6dGFibGUtY2VsbFwiPk5hbWU8L2Rpdj5cbiAgICAgIDxkaXYgc3R5bGU9XCJkaXNwbGF5OnRhYmxlLWNlbGxcIj5WaWV3czwvZGl2PlxuICAgICAgPGRpdiBzdHlsZT1cImRpc3BsYXk6dGFibGUtY2VsbFwiPldyaXRlcjwvZGl2PlxuICAgICAgPGRpdiBzdHlsZT1cImRpc3BsYXk6dGFibGUtY2VsbFwiPkNyZWF0aW9uVGltZTwvZGl2PlxuICAgIDwvZGl2PlxuICAgIDxleGFtcGxlLXJvdy1pdGVtIGNvbnRlbnROYW1lPVwiTGl0IGlzIGdvb2RcIiB2aWV3cz1cIjQ0NTJcIiB3cml0ZXI9XCJ0ZXN0XCIgY3JlYXRpb25UaW1lPVwiMjAyMi0xMS0zMCAxMjoyNDozMFwiPjwvZXhhbXBsZS1yb3ctaXRlbT5cbiAgICA8ZXhhbXBsZS1yb3ctaXRlbSBjb250ZW50TmFtZT1cIlRoaXJkIEl0ZW1cIiB2aWV3cz1cIjIyNTNcIiB3cml0ZXI9XCJ0ZXN0XCIgY3JlYXRpb25UaW1lPVwiMjAyMi0xMS0zMCAxMjoyNDozMFwiPjwvZXhhbXBsZS1yb3ctaXRlbT5cbiAgICA8ZXhhbXBsZS1yb3ctaXRlbSBjb250ZW50TmFtZT1cIlNlY29uZCBDb250ZW50XCIgdmlld3M9XCIxMzM3XCIgd3JpdGVyPVwidGVzdFwiIGNyZWF0aW9uVGltZT1cIjIwMjItMTEtMzAgMTI6MjQ6MzBcIj48L2V4YW1wbGUtcm93LWl0ZW0-XG4gICAgPGV4YW1wbGUtcm93LWl0ZW0gY29udGVudE5hbWU9XCJGaXJzdCBJdGVtXCIgdmlld3M9XCIzMzA2XCIgd3JpdGVyPVwidGVzdFwiIGNyZWF0aW9uVGltZT1cIjIwMjItMTEtMzAgMTI6MjQ6MzBcIj48L2V4YW1wbGUtcm93LWl0ZW0-XG4gIDwvZGl2PlxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ)
